gtsam_unstable/discrete/tests/testCSP.cpp

```diff
@@ -120,8 +120,8 @@ TEST_UNSAFE(CSP, WesternUS) {
   Ordering ordering;
   ordering += Key(0), Key(1), Key(2), Key(3), Key(4), Key(5), Key(6), Key(7),
       Key(8), Key(9), Key(10);
-  CSP::sharedValues mpe = csp.optimalAssignment(ordering);
-  // GTSAM_PRINT(*mpe);
+  auto mpe = csp.optimalAssignment(ordering);
+  // GTSAM_PRINT(mpe);
   CSP::Values expected;
   insert(expected)(WA.first, 1)(CA.first, 1)(NV.first, 3)(OR.first, 0)(
```

Comment:
Sorry for jumping in, but I remember last time you asked me to wrap some discrete in Python, and I realized this is not easy to wrap. Could we drop this syntax (which is boost specific) and use some C++11-only style instead?

Reply:
different PR